### DISCUSSION TITLE
Restore admin account

### DISCUSSION THREAD

**Discussion by MrMarble** [seerr-team/seerr#345] [Q&A]
I have both _jellyfin_ and _jellyseerr_ running on **docker**, somehow (maybe during an update of *jellyfin*) the tokens for the users changed and when trying to log in *jellyseerr* with my admin account (which is my only account, synced with jellyfin), it creates a new user account, keeping me out of the "*owner*" one.

I can see some sync errors stating an auth error when trying to sync with jellyfin:

```yaml
2023-03-13T10:00:00.066Z [error][Jellyfin API]: Something went wrong while getting library content from the Jellyfin server: Request failed with status code 401 
2023-03-13T10:00:00.066Z [error][Jellyfin Sync]: Sync interrupted {"errorMessage":"Invalid auth token"}
```

peeking through the `db.sqlite3` file, I can see that a new user is created with the same name:

```sql
select id, permissions, userType, jellyfinUsername, jellyfinAuthToken, createdAt from user;                
```
Output (pretty printed for easy of reading):

|id|permissions|userType|jellyfinUsername|jellyfinAuthToken|createdAt|
|--|--|--|--|--|--|
|1|2|3|mrmarble|440e72XXXXXXXXXXXXXXXXXXXXXXXXXX|2022-10-22 16:53:24|
|2|32|3|mrmarble|1dba8cXXXXXXXXXXXXXXXXXXXXXXXXXX|2023-03-13 10:23:37|

I think jellyseerr should use the same user updating the token instead of creating a new one.

**Comment by fallenbagel** [comment]
It does use the same user ehen updating token but im not sure why yours did that. Could be a little db issue. You could try one of two things. Restore from a backup (assuming you restored) or edit sqlite and change your new user to owner and remove old owner

**Reply by MrMarble** [reply]
I tried deleting the new user and login again but the same happened. Will try later to manually change the `jellyfinAuthToken` and if that doesn't works, I'll remove the user and update the new one.

**Reply by fallenbagel** [reply]
Do let me know how it goes. I'm interested to narrow down why this happened. You're probably the first person who had this issue in the past year

**Reply by MrMarble** [reply]
Having `wal` enabled is causing a bit of pain for testing... (had to do a bunch of changes until the page size went over the 1k limit for SQLite to write to the db)

I fixed it by overwriting `jellyfinAuthToken`, `jellyfinUserId`, `jellyfinDeviceId` from the new user to the original "owner" and deleting the latest user, that way I can keep all the history of my requests.

Not sure how or why got this bug tbh

**Comment by sebinum** [comment]
Just wanted to share that I've run into the same issue. Seems that with the last update (from 1.2 to 1.4) it created another user with the same sync issues shared by @MrMarble.  As I wasn't bothered about my request history too much I did a quick re-install.

**Reply by fallenbagel** [reply]
Could be due to the huge jump from 1.2 to 1.4 without going through 1.2.x and 1.3 and 1.3.x

**Comment by dreams-and-thoughts** [comment]
Accidentally replied to wrong comment.

How do I go about changing a user to Admin/Owner?

I think I have the same, or similar issue to the OP. I'm getting this warn in the logs, with only a single user:

`﻿[warn][Jellyfin Sync]: No admin configured. Jellyfin sync skipped.`

And the users table:

| email | username | permissions | userType | jellyfinUsername | jellyfinAuthToken      
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| meee |  | 32 | 3 | meee | e5fdd90a5XXX |

There were 2x users, with the same Jellyfin username. The other user had permissions '2' and userType '3'. I deleted that user thinking it was the issue.

Everything was previously working. The only thing I can think that might be responsible is that the Jellyfin server changed ports, and then JellySeer was unable to connect. Maybe after reconnecting and authenticating the user changed?

**Reply by fallenbagel** [reply]
So ID=1 is your owner. You just have to change the jellyfinUserId of that user to your current one (you can find your current jellyfinUserId as the one thays duplicated last. Then delete the rest).

This happened because jellyfin for some reason changed your jellyfin userid (which is not supposed to change)
